core/jvm/src/test/scala/fs2/MemorySanityChecks.scala

```diff
@@ -52,24 +52,24 @@ object ResourceTrackerSanityTest extends App {
 }
 
 object RepeatPullSanityTest extends App {
-  def id[A]: Pipe[Pure, A, A] = _.repeatPull {
+  def id[F[_], A]: Pipe[F, A, A] = _.repeatPull {
     _.uncons1.flatMap {
       case Some((h, t)) => Pull.output1(h).as(Some(t));
       case None         => Pull.pure(None)
     }
   }
-  Stream.constant(1).covary[IO].through(id[Int]).compile.drain.unsafeRunSync()
+  Stream.constant(1).covary[IO].through(id[IO, Int]).compile.drain.unsafeRunSync()
 }
 
 object RepeatEvalSanityTest extends App {
-  def id[A]: Pipe[Pure, A, A] = {
-    def go(s: Stream[Pure, A]): Pull[Pure, A, Unit] =
+  def id[F[_], A]: Pipe[F, A, A] = {
```

Comment:
Did these have to change? If so, could you update the PR with a summary of public facing source compatibility API changes?

Reply:
So, at some point in writing these changes, I thought I may need to change it. It 

Now that the `F` is covariant and not invariant, I am not sure if there is still much use in keeping the idea of a `Pipe` with a same parameter for the input and output stream. Ideally, you may want the pipe to be applied to any sub-F-type of the input, and be compatible with any super-F-type of the output (just like any function).

Reply:
`F` in `Stream` and `Pull` have always been covariant though, so changing internals shouldn't cause changes to the API/usage.